packages/linter/src/cli.ts

```diff
@@ -0,0 +1,238 @@
+      "override output format",
+      /^(text|json|tsv|html)$/i,
+      "text"
+    )
+    .option(
+      `-A, --user-agent <string>`,
+      "user agent string",
+      /^(googlebot_desktop|googlebot_mobile|chrome_desktop|chrome_mobile)$/i,
+      "googlebot_mobile"
+    )
+    .on("--help", function() {
+      console.log("");
+      console.log("Examples:");
+      console.log("  $ amplint https://www.ampproject.org/");
+      console.log("  $ amplint --force sxg https://www.ampbyexample.org/");
+    });
+
+  if (argv.length <= 2) {
+    program.help();
+  }
+
+  function seq(first: number, last: number): number[] {
+    if (first < last) {
+      return [first].concat(seq(first + 1, last));
+    } else if (first > last) {
+      return [last].concat(seq(first, last - 1));
+    } else {
+      return [first];
+    }
+  }
+
+  // One reason to support curl-style arguments is to provide cookies that avoid
+  // GDPR interstitials.
+  const headers: { [k: string]: string } = seq(2, argv.length - 1)
+    .filter(n => argv[n] === "-H")
+    .map(n => argv[n + 1])
+    .map(s => {
+      const [h, ...v] = s.split(": ");
+      return [h, v.join("")];
+    })
+    .reduce((a: { [key: string]: any }, kv) => {
+      a[kv[0]] = kv[1];
+      return a;
+    }, {});
+
+  // Options is argv with "curl" and all -H flags removed (to pass to
+  // program.parse())
+  const options = seq(0, argv.length - 1)
+    .filter(n => argv[n] !== "curl" && argv[n] !== "-H" && argv[n - 1] !== "-H")
+    .map(n => argv[n]);
+
+  program.parse(options);
+
+  const url = program.args[0];
+  if (!url) {
+    program.help();
+  } else {
+    program.url = url;
+  }
+
+  program.headers = headers;
+
+  return easyLint((program as unknown) as {
+    userAgent: string;
+    format: string;
+    force: LintMode | "auto";
+    url: string;
+    headers: { [k: string]: string };
+  })
+    .then(console.log)
+    .catch(e => {
+      console.error(e.stack || e.message || e);
+      process.exitCode = 1;
+    });
+}
+
+export function easyLint({
+  url,
+  userAgent,
+  format,
+  force,
+  headers
+}: {
+  url: string;
+  userAgent: string;
+  format: string;
+  force: LintMode | "auto";
+  headers: { [k: string]: string };
+}) {
+  headers["user-agent"] = UA[userAgent as keyof typeof UA];
+
+  const raw = (async () => {
+    if (url === "-") {
+      return Promise.resolve({
+        body: readFileSync("/dev/stdin").toString(),
+        headers: {}
+      });
+    }
+    const debug = fetchToCurl(url, { headers });
+    try {
+      const res = await fetch(url, { headers });
+      return res.ok
+        ? Promise.resolve({
+            headers: res.headers,
+            body: await res.text()
+          })
+        : Promise.reject(
+            `couldn't load [${url}]: ${res.statusText} [debug: ${debug}]`
+          );
+    } catch (e) {
+      return Promise.reject(`couldn't load [${url}] [debug: ${debug}]`);
+    }
+  })();
+
+  return raw
+    .then(r => {
```

Comment:
why not use `await`?

Reply:
Simplified in a1403a9f7810ce5af832b9d31a36e8af643c1ccc.